laika/downloader.py

```diff
@@ -198,12 +196,14 @@ def https_download_file(url):
 
 
 def ftp_download_file(url):
+  parsed = urlparse(url)
   try:
-    urlf = urllib.request.urlopen(url, timeout=10)
-    data_zipped = urlf.read()
-    urlf.close()
-    return data_zipped
-  except urllib.error.URLError as e:
+    buf = BytesIO()
```

Comment:
Maybe collapse to a single function where you look at the scheme of the url to determine whether to use the FTP or FTPS class since the code looks very similar?